automated install/basic-install.sh

```diff
@@ -1973,8 +2038,104 @@ accountForRefactor() {
         if grep -q '^INSTALL_WEB_INTERFACE=true' ${setupVars}; then
             webserver_installed=true
         fi
-        echo -e "INSTALL_WEB_SERVER=$webserver_installed" >> "${setupVars}"
+        replaceAddCreateRename INSTALL_WEB_SERVER webserver_installed "${setupVars}"
+    fi
+}
+
+# A function that validates the pihole_setupVars.conf required for unattended installation and updates.
+validate_setupVars () {
+    printf "  Validating setupVars.conf\\n"
+    echo ""
+    # Replace any crlf with nl if they exists in the source file
+    sed -i 's/\r//'  ${setupVars}
+    # Does setupVars contain the minimume variables?
+    local requiredVars=(PIHOLE_INTERFACE IPV4_ADDRESS PIHOLE_DNS_1 PIHOLE_DNS_2)
+    # Variables with true or false as the only options
+    local optionalVarsTF=(QUERY_LOGGING INSTALL_WEB_SERVER INSTALL_WEB_INTERFACE BLOCKING_ENABLED DNS_FQDN_REQUIRED DNS_BOGUS_PRIV DNSSEC CONDITIONAL_FORWARDING)
+    source ${setupVars}
+    for var in "${requiredVars[@]}"; do
+        if [[ -z "${!var}" ]]; then
+            echo "${var} is not set"
+            exit 1
+        else
+            echo "${var} is ${!var}"
+        fi
+    done
+    for i in "${optionalVarsTF[@]}"; do
+        if [[ "${!var}" == "true" ]] || [[ "${!var}" == "false" ]]; then
+            echo "${var} is ${!var}"
+        elif [[ -z "${!var}" ]]; then
+            echo "${var} is not set"
+        else
+            echo "${var} is ${!var}"
+            echo "If ${var} is set in setupVars.conf, valid options are: true or false."
+            exit 1
+        fi
+    done
+    # INSTALL_WEB_SERVER must be set to true if INSTALL_WEB_INTERFACE is set to true
+    if [[ "${INSTALL_WEB_INTERFACE}" == true ]] && [[ "${INSTALL_WEB_SERVER}" == false ]]; then
+        echo "You can install the Web Server without the web interface, but you cannot install the web interface without the web server."
+        echo "INSTALL_WEB_SERVER is set to ${INSTALL_WEB_SERVER} and INSTALL_WEB_INTERFACE is set to ${INSTALL_WEB_INTERFACE}."
+        exit 1
+    fi
+    # Regex patterns to match
```

Comment:
No need to reinvent the wheel here, please see (and re-use) the existing functions `valid_ip` and `valid_ip6` (there is a change to `valid_ip6` in the following [commit](https://github.com/pi-hole/pi-hole/pull/3564/commits/643d2c0f3ea7871b64f9eaa2c9a5a2060d9ad14e) in #3564)

Reply:
The IP address has to include the subnet mask.  I looked at those functions but they do not check for subnet in bits "/xx".  
I can remove line 2084, 'addressPort' and replace any uses with valid_ip.
Line 2082, 'address' is not used currently so I can remove it.

Line 2083 and 2085, 'addressMask' and 'addressReverse' need to stay.  The existing functions won't validate them.